subscription-engine/kafka-ws/app/WSServer.js

```diff
@@ -0,0 +1,107 @@
+const { logger } = require('@dojot/dojot-module-logger');
+
+const WebSocket = require('ws');
+const url = require('url');
+
+const { WSError } = require('./Errors').Errors;
+const { ErrorCodes } = require('./Errors');
+const { ProcessingRuleManager } = require('./ProcessingRule');
+const { WhereParser } = require('./WhereProcessing');
+
+const TAG = { filename: 'WSServer' };
+
+
+/**
+  * WebSocket Server.
+  *
+  * @param {http.Server} server HTTP server to be used
+  * @param {string} path URL path that will accept WebSocket connections
+  */
+class WSServer {
+  constructor() {
+    logger.info('Initializing WebSocket Server...', TAG);
+    // We initialize with `noServer: true` so we can add this server to a specific path in the HTTP
+    // server
+    this.wsServer = new WebSocket.Server({ noServer: true });
```

Comment:
To use the websocket with security we would have to do something like this in the code:

```js
const server = https.createServer({
    cert: fs.readFileSync('/opt/testws/certs/wscert.crt'),
    key: fs.readFileSync('/opt/testws/certs/wscert.key'),
    ca: [fs.readFileSync('/opt/testws/certs/ca.crt')],
    rejectUnauthorized: true,
    requestCert: true
});
const wss = new WebSocket.Server({ server })
```

it must be possible to configure the paths for `cert`,` key` and `ca` by the environment variables.

It would also be important to be able to enable or disable this secure mode, by default I believe it should be disabled, "false".
I don't know if it would be nice if you put these settings in now, or I put them in the future, as you prefer.

What do you think? @rascaraficci @eduardogmisiuk

Reply:
Another example:
```js 
const fs = require('fs');
const https = require('https');

https
  .createServer(
    {
    cert: fs.readFileSync('/opt/testws/certs/wscert.crt'),
    key: fs.readFileSync('/opt/testws/certs/wscert.key'),
    ca: [fs.readFileSync('/opt/testws/certs/ca.crt')],
    },
    (req, res) => {
      if (!req.client.authorized) {
        res.writeHead(401);
        return res.end('Invalid client certificate authentication.');
      }

      res.writeHead(200);
      res.end('Hello, world!');
    }
  )
  .listen(9443);
```

Reply:
Yes, I think we could add the secure mode in te future too. We could wait until we finish the main functionality of the service.

Another thing I was thinking about was to remove Express. We don't use a lot of functionalities it gives us and the http/https modules are faster. Originally, I made the service with HTTPServer, but changed to Express (hence the comment in WSServer.js:23 that I should've removed).

Reply:
Complementing [mprevide's](https://github.com/mprevide) comment, the flags `rejectUnauthorized: true|false` and `requestCert: true|false` would have to be set via environment variables, as they are directly related to the Kong _proxy_ settings.
It would be ideal if the same environment variables that enable SSL verification on the Kong proxy side are also used to activate these flags on this service.